qa/rolling-upgrade/src/test/java/org/opensearch/upgrades/IndexingIT.java

```diff
@@ -202,24 +193,297 @@ public void testAutoIdWithOpTypeCreate() throws IOException {
         }
     }
 
+    public void testTypeRemovalIndexing() throws IOException {
+        final String indexName = "test_index_with_mapping";
+        final String indexWithoutTypeName = "test_index_without_mapping";
+        final String templateIndexName = "template_test_index";
+        final String templateName = "test_template";
+        final String indexNamePattern = "template_test*";
+
+        final String mapping = "\"properties\":{\"f1\":{\"type\":\"keyword\"},\"f2\":{\"type\":\"keyword\"}}";
+        final String typeMapping = "\"_doc\":{" + mapping +"}";
+
+        switch (CLUSTER_TYPE) {
+            case OLD:
+                Version minNodeVersion = getMinNodeVersion();
+                if (minNodeVersion.before(Version.V_2_0_0)) {
```

Comment:
When all the nodes in the cluster is in the same version, I think it's easier to directly use the variable `UPGRADE_FROM_VERSION`. (Please see the other test classes in the directory https://github.com/opensearch-project/OpenSearch/blob/1.3.1/qa/rolling-upgrade/src/test/java/org/opensearch/upgrades/JodaCompatibilityIT.java#L77 and https://github.com/opensearch-project/OpenSearch/blob/1.3.1/qa/rolling-upgrade/src/test/java/org/opensearch/upgrades/MappingIT.java#L45)

In addition, I think getting the `minNodeVersion` is originally designed to be used in a mixed-version cluster (`case MIXED`), when an API response from nodes of different versions are different. My conclusion is derived from the original test case `testAutoIdWithOpTypeCreate()` (https://github.com/opensearch-project/OpenSearch/blob/1.3.1/qa/rolling-upgrade/src/test/java/org/opensearch/upgrades/IndexingIT.java#L211-L213)

Further, learning from the above 2 tests classes, probably it will be more concise to use `assumeTrue` to filter the versions to run the test. 😁
```
assumeTrue("Mapping types is removed in 2.0", UPGRADE_FROM_VERSION.beforeVersion.V_2_0_0));
```

Reply:
I was leaning on the `minNodeVersion` check to avoid any race/weird condition before I actually push in the typed indices. It does not seem to be an issue in my tests. Moved it to a regular check.

`assertTrue` will not work for all the cases since the `bwcTest` runs checks between not just current branch but a bunch of versions depending on compatibility. (https://github.com/opensearch-project/OpenSearch/blob/1b571ece28f5644becf97c165de853943af73ac5/qa/rolling-upgrade/build.gradle#L40) Backporting to older branches will break the assumption.

Reply:
Ah, got your reason to use `minNodeVersion` check 😄 👍. The current code looks good, but I need some time to understand the limitation of `assumeTrue` you pointed out.

Reply:
The meaning of `assumeTrue("...", UPGRADE_FROM_VERSION.beforeVersion.V_2_0_0);` is to filter the test to only run when upgrading from a cluster < 2.0 version.
I just learnt this from seeing the other test class, such as https://github.com/opensearch-project/OpenSearch/blob/1.3.1/qa/rolling-upgrade/src/test/java/org/opensearch/upgrades/JodaCompatibilityIT.java#L77

You could see the javadoc of `assumeTrue()` (https://javadoc.io/doc/com.carrotsearch.randomizedtesting/randomizedtesting-runner/2.7.1/com/carrotsearch/randomizedtesting/RandomizedTest.html#assumeTrue-java.lang.String-boolean-)
> condition - If false an AssumptionViolatedException is thrown by this method and the test case (should be) ignored. Tests that are assumption-failures do not break builds.

In my opinion, breaking the assumption is the result we want. 😄

Reply:
I misread it as `assertTrue`! This is a great find. Thanks for linking this 🙂 
The if conditions currently do cover all of over scenarios, but will update with `assumeTrue` whenever I push any changes out for the test.

Reply:
Sounds good! Yeah.. your PR does a good opportunity for me to find those interesting usages.